src/corva/models/scheduled/raw.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+import itertools
+from typing import List, Union
+
+import pydantic
+
+from corva.api import Api
+from corva.models.base import CorvaBaseEvent, RawBaseEvent
+from corva.models.scheduled.initial import InitialScheduledEvent
+from corva.models.scheduled.scheduler_type import SchedulerType
+
+
+def set_schedule_start(schedule_start: int) -> int:
+    """Casts Unix timestamp from milliseconds to seconds.
+
+    Casts Unix timestamp from millisecond to seconds, if provided timestamp is in
+        milliseconds.
+    """
+
+    # 1 January 10000 00:00:00 - first date to not fit into the datetime instance
+    if schedule_start >= 253402300800:
+        schedule_start //= 1000
+
+    return schedule_start
+
+
+class RawScheduledEvent(CorvaBaseEvent, RawBaseEvent):
+    """Base class for raw scheduled event data.
+
+    Attributes:
+        asset_id: asset id.
+        company_id: company id.
+        interval: depth or time between two schedule triggers.
+        schedule_id: schedule id.
+        app_connection_id: app connection id.
+        app_stream_id: app stream id.
+        scheduler_type: type of scheduled event.
+    """
+
+    asset_id: int
+    company_id: int = pydantic.Field(..., alias='company')
+    interval: float
+    schedule_id: int = pydantic.Field(..., alias='schedule')
+    app_connection_id: int = pydantic.Field(..., alias='app_connection')
+    app_stream_id: int = pydantic.Field(..., alias='app_stream')
+    scheduler_type: SchedulerType
+
+    @staticmethod
+    def from_raw_event(event: Union[dict, List[List[dict]]]) -> List[RawScheduledEvent]:
+        if isinstance(event, dict):
+            event = [[event]]
+
+        # flatten the event into 1d array
+        event: List[dict] = list(itertools.chain(*event))
+
+        initial_events = pydantic.parse_obj_as(List[InitialScheduledEvent], event)
+
+        events = [
+            initial_event.scheduler_type.raw_event.parse_obj(sub_event)
+            for initial_event, sub_event in zip(initial_events, event)
+        ]
+
+        return events
+
+    def set_schedule_as_completed(self, api: Api) -> None:
+        """Sets schedule as completed."""
+        api.post(path=f'scheduler/{self.schedule_id}/completed')
```

Comment:
Will this retry on failure?

Reply:
No, there is no retries in Api class

Reply:
We should probably allow users to provide their own transport or something to easily enable retries